### PR TITLE
Inform to sign in for deploy if not signed in

### LINKED ISSUE #472: Let user know that he/she needs to log in for deploy
When the user is not logged in and wants to deploy to App Engine, the plugin simply pops up a browser with the authentication page without any explanation. We should tell the user the reason for the login page.

### PR DESCRIPTION
**Pull request by chanseokoh**
Fixes #472 #524.

prompt-login-for-deploy

Moved out `login()` from `launchDeployJob()`, since `launchDeployJob()` sounds like it will launch a job no matter what.

Did some tests: cancel login, deny authorization, deploy while logged in, etc.

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/messages.properties [hunk 1]
```diff
@@ -10,5 +10,6 @@ deploy.standard.runnable.name=Deploying to App Engine Standard
 deploy.job.stagingdir.create.failed=Cannot create staging directory.
 deploy.job.sourcedir.missing=Staging source directory does not exist.
 deploy.failed.error.message=Deploy failed.
-login.failed=Login failed.
-save.credential.failed=Saving credential to file failed.
\ No newline at end of file
+save.credential.failed=Saving credential to file failed.
+prompt.login.dialog.title=Sign-in Required for Deploying
+prompt.login.dialog.message=Do you want to sign in to authorize deploying to the target project?
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/messages.properties]: Do you want to sign in to deploy the target project?

**Comment by elharo** [comment]
per #472 "We should tell the user the reason for the login page."

I don't think this means we should ask them if they want to sign in. Assume they do. Ideally the browser page that pops up should include the message. However if that's difficult or infeasible a simple message dialog (Please sign in to deploy. OK) will do.

**Comment by chanseokoh** [comment]
Customizing the title of the login page doesn't seem possible: https://developers.google.com/identity/protocols/OAuth2InstalledApp

I'll change the question dialog with a simple message dialog just with a OK button. (Alternatively, it is also possible to change the title of the login progress monitor dialog. However, since the external login browser will pop up and get the focus, a simple message dialog before launching the browser seems a good idea.)

**Comment by chanseokoh** [comment]
The dialog looks like this now:

sign-in-to-deploy

**Comment by patflynn** [comment]
This wording doesn't sound ideal to me.  Looking at it I'd wonder 'They
want me to open an external browser and login...how?  Go to google.com? How
does that authenticate the IDE?"

**Comment by chanseokoh** [comment]
@patflynn Right... I'm too accustomed to our login flow.

**Comment by elharo** [comment]
They don't have to press the button to get the browser, right? It just opens automatically. So here we just want a statement, not a call to action. E.g. 

"Browser opened to authorize deployment." 
OK

**Comment by chanseokoh** [comment]
So, the last commit just customizes the message of the default login progress dialog. (No extra dialog; a browser will open automatically for deployment if users are not logged in.)

login-progress-dialog

**Comment by elharo** [comment]
LGTM iff the tests pass on travis

**Pull request merged**